### PR TITLE
Creating types for record array proxy

### PR DESCRIPTION
**Pull request by jasonmit**
Moving record array and filtered record array out into their own types.  We should also look to create through the new `getOwner` API.

### CODE REVIEW

**Comment by toranb** [comment]
Before we merge all these in should we mere the v4 work into master? If not I don't imagine a clean rebate after you get done fixing all my stylistic issues :)

**Comment by jasonmit** [comment]
I'm all for that, any more work on 3.x can just be branches from previous release tags and cherry picked back into master if they still apply.  Maybe just throw up a disclaimer on the README pointing to the latest stable README

**Comment by toranb** [comment]
@jasonmit sounds good - just threw up the v4.0.0 PR and if travis says "g2g" I'll merge it in so you can rebase these 2 PR's

**Comment by toranb** [comment]
How backward compatible is the work you mention around `getOwner` ? I'm assuming 1.13 people could use this right?

**Comment by toranb** [comment]
all good for the rebase now - hope it's not terribly painful

**Comment by jasonmit** [comment]
@toranb I branched from v4 so nothing needed on my end just needed to target master instead of v4.  This PR should be merged first though since it's going into my other branch.

File: addon/store.js [hunk 1]
```diff
@@ -65,7 +70,7 @@ var Store = Ember.Object.extend({
     push(type, data) {
         var record = this._findById(type, data.id);
         if (record) {
-            record.setProperties(data);
+            setProperties(record, data);
```
Comment (jasonmit) [inline addon/store.js:73]: Eventually opening up the possibility to support POJOs.  I removed the model.get in my other branch.

**Comment by toranb** [comment]
@jasonmit cool cool - go ahead and merge it when ready boss

File: addon/store.js (hunk 1 above)
Comment (toranb) [inline addon/store.js:73]: that sounds awesome - good work my man

**Pull request merged**